src/basic/missing_syscalls.py

```diff
@@ -75,8 +75,9 @@ def parse_syscall_tables(filenames):
 #    else
 #      define systemd_NR_{syscall} {nr_x86_64}
 #    endif
-#  else
-#    warning "{syscall}() syscall number is unknown for your architecture"
+#  elif !defined(systemd_warned_syscall)
+#    warning "syscall numbers are unknown for your architecture"
+#    define systemd_warned_syscall 1
```

Comment:
is this right? new syscalls dded to the upstream kernel only appear on some archs first. they are simply not defined on other archs then. the whole huge ifdeffery here can't really cope for that right now, not sure we really should make it more complex.

Reply:
I think the practice has changed. All new syscalls are added "everywhere", and the implementation is hooked up to various architectures over time. So instead of syscall numbers being unpredictable, for new syscalls it's some number + constant offset for other architectures… https://github.com/torvalds/linux/commit/2a1867219c7b27f928e2545782b86daaf9ad50bd, https://github.com/torvalds/linux/commit/b0a0c2615f6f199a656ed8549d7dce625d77aa77, https://github.com/torvalds/linux/commit/ecb8ac8b1f146915aa6b96449b66dd48984caacc. I'm not sure if we should count on that though.

/cc @hrw

Reply:
Anyway, I'll rework this to be a bit more smart.

Reply:
hmm, but if it's actually the same now everywhere, then how come we still have these ifdef monsters even for the new syscalls? there must be some better way...

i.e. in #19438 I add some new syscalls, and quite frankly it sucks I have to add the syscalls there for all archs now the CI runs on, given i have zero understanding of the mappings on those archs. i#d much rather leave that for people to add who actually care about these archs..

Reply:
The "ifdef monster" is just because it's easier (not more hard?) to generate it this way. We have a check for `!defined(__IGNORE_{syscall})`, then per-arch checks nested inside. We could invert that. If this was written by hand, it could maybe be better. But since now it's generated, it doesn't matter.

Hmm, did you do the addition to  `missing_syscall_def.h` by hand? I'll add it to this PR autogenerated instead.

Reply:
yes, i did. i wasn't aware it was auto-generated, and no idea how i'd add something there then?